packages/sources/lotus/src/transport/balance.ts

```diff
@@ -0,0 +1,141 @@
+import { TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+import { AdapterResponse, sleep } from '@chainlink/external-adapter-framework/util'
+import { BaseEndpointTypes, inputParameters } from '../endpoint/balance'
+import { BigNumber } from 'ethers'
+import { calculateHttpRequestKey } from '@chainlink/external-adapter-framework/cache'
+import { SubscriptionTransport } from '@chainlink/external-adapter-framework/transports/abstract/subscription'
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+
+export type BalanceTransportTypes = BaseEndpointTypes & {
+  Provider: {
+    RequestBody: never
+    ResponseBody: {
+      method: string
+      params: string[]
+      id: number
+      jsonrpc: string
+    }
+  }
+}
+
+type RequestParams = typeof inputParameters.validated
+
+export class TotalBalanceTransport extends SubscriptionTransport<BalanceTransportTypes> {
+  requester!: Requester
+  settings!: BalanceTransportTypes['Settings']
+  endpointName!: string
+
+  async initialize(
+    dependencies: TransportDependencies<BalanceTransportTypes>,
+    _adapterSettings: BalanceTransportTypes['Settings'],
+    _endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, _adapterSettings, _endpointName, transportName)
+    this.requester = dependencies.requester
+    this.settings = _adapterSettings
+    this.endpointName = _endpointName
+  }
+
+  async backgroundHandler(
+    context: EndpointContext<BalanceTransportTypes>,
+    entries: RequestParams[],
+  ) {
+    if (!entries.length) {
+      await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+      return
+    }
+    await Promise.all(entries.map(async (param) => this.handleRequest(param)))
+  }
+
+  async handleRequest(param: RequestParams) {
+    let response: AdapterResponse<BaseEndpointTypes['Response']>
+    try {
+      response = await this._handleRequest(param)
+    } catch (e) {
+      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
+      response = {
+        statusCode: 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: 0,
+          providerDataReceivedUnixMs: 0,
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: param, response }])
+  }
+
+  async _handleRequest(
+    param: RequestParams,
+  ): Promise<AdapterResponse<BalanceTransportTypes['Response']>> {
+    const addresses = param.addresses
+
+    const providerDataRequestedUnixMs = Date.now()
+
+    const balances = await Promise.all(
+      addresses.map((addr, index) => this.getBalance(addr.address, index)),
+    )
+
+    const result = balances
+      .reduce((sum, balance) => sum.add(balance.result as unknown as BigNumber), BigNumber.from(0))
```

Comment:
We can maybe just initialized the BigNumber here to make this cleaner. If that gives you an error, you may need to turn `balance.result` into a string first
```suggestion
      .reduce((sum, balance) => sum.add(BigNumber(balance.result)), BigNumber.from(0))
```